### DISCUSSION TITLE
Fatal error: Python.h: No such file or directory

### DISCUSSION THREAD

**Discussion by bmahlbrand** [community/community#27036] [Actions]
<p>I’m wondering how to manage my python 3.8 environment in github actions, I’m using the following in my action, but I can’t seem to get the headers into the PATH.</p>
<p>fatal error: Python.h: No such file or directory</p>
<p><a href="https://github.com/user/repo/runs/2968137274?check_suite_focus=true#step:4:332" rel="noopener nofollow ugc">332</a> 124 | <span class="hashtag">#include</span> &lt;Python.h&gt;</p>
<p><a href="https://github.com/user/repo/runs/2968137274?check_suite_focus=true#step:4:333" rel="noopener nofollow ugc">333</a></p>
<blockquote>
<pre><code>  # Sets up python3
  - uses: actions/setup-python@v2
    with:
      python-version: 3.8 

  # Installs and upgrades pip, installs other dependencies and installs the package from setup.py
  - name: "Installs and upgrades pip, installs other dependencies and installs the package from setup.py"
    run: |
      # dev headers
      sudo apt-get install python3.8-dev  # for python3.x installs
      sudo apt-get install libpython3.8-dev
      # Upgrade pip
      python3 -m pip install --upgrade pip
      # Install build deps
      python3 -m pip install setuptools wheel twine
      # If requirements.txt exists, install from it
      if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
      # Install the package from setup.py
      python3 setup.py install</code></pre>
</blockquote>

**Comment by airtower-luna** [comment]
<p>It looks like you’re mixing the Python version set up by <code>actions/setup-python</code> with one from the Ubuntu repositories. <code>python3</code> should be the version from Ubuntu (<code>actions/setup-python</code> defines what <code>python</code> is), but this line uses <code>pip</code>, not the <code>pip</code> module available to <code>python3</code>:</p>
<aside class="quote no-group" data-username="bmahlbrand" data-post="1" data-topic="188171">
<div class="title">
<div class="quote-controls"></div>
 bmahlbrand:</div>
<blockquote>
<pre><code class="lang-auto">      if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
</code></pre>
</blockquote>
</aside>
<p>I don’t know which version is better for your project, but mixing probably isn’t good. I’m not sure if <code>actions/setup-python</code> provides the C headers, if not you might have to use the Ubuntu version.</p>

**Comment by bmahlbrand** [comment, marked as answer]
<p>It turns out I had a non portable path set up in my cmakelists.txt that setup.py was running. To confirm for future travelers, just using actions/setup-python@v2 should have the C headers installed in /usr/include/python3.8 (in this case) should work as expected, the ubuntu headers aren’t required, so I deleted them.</p>
